src/opentime/rationalTime.h

```diff
@@ -121,55 +164,69 @@ class RationalTime {
     }
 
-    friend RationalTime operator+ (RationalTime lhs, RationalTime rhs) {
+    friend RationalTime operator+ (const RationalTime& lhs, const RationalTime& rhs) {
         return (lhs._rate < rhs._rate) ? RationalTime {lhs.value_rescaled_to(rhs._rate) + rhs._value, rhs._rate} :
                                          RationalTime {rhs.value_rescaled_to(lhs._rate) + lhs._value, lhs._rate};
     }
         
-    friend RationalTime operator- (RationalTime lhs, RationalTime rhs) {
+    friend RationalTime operator- (const RationalTime& lhs, const RationalTime& rhs) {
         return (lhs._rate < rhs._rate) ? RationalTime {lhs.value_rescaled_to(rhs._rate) - rhs._value, rhs._rate} :
                                          RationalTime {lhs._value - rhs.value_rescaled_to(lhs._rate), lhs._rate};
     }
 
-    friend RationalTime operator- (RationalTime lhs) {
+    friend RationalTime operator- (const RationalTime& lhs) {
         return RationalTime {-lhs._value, lhs._rate};
     }
 
-    friend bool operator> (RationalTime lhs, RationalTime rhs) {
+    friend bool operator> (const RationalTime& lhs, const RationalTime& rhs) {
          return (lhs._value / lhs._rate) > (rhs._value / rhs._rate);
     }
 
-    friend bool operator>= (RationalTime lhs, RationalTime rhs) {
+    friend bool operator>= (const RationalTime& lhs, const RationalTime& rhs) {
         return (lhs._value / lhs._rate) >= (rhs._value / rhs._rate);
     }
 
-    friend bool operator< (RationalTime lhs, RationalTime rhs) {
+    friend bool operator< (const RationalTime& lhs, const RationalTime& rhs) {
         return !(lhs >= rhs);
     }
 
-    friend bool operator<= (RationalTime lhs, RationalTime rhs) {
+    friend bool operator<= (const RationalTime& lhs, const RationalTime& rhs) {
         return !(lhs > rhs);
     }
 
-    friend bool operator== (RationalTime lhs, RationalTime rhs) {
+    friend bool operator== (const RationalTime& lhs, const RationalTime& rhs) {
         return lhs.value_rescaled_to(rhs._rate) == rhs._value;
     }
 
-    friend bool operator!= (RationalTime lhs, RationalTime rhs) {
+    friend bool operator!= (const RationalTime& lhs, const RationalTime& rhs) {
         return !(lhs == rhs);
     }
 
-private:
-    static RationalTime _invalid_time;
-    static constexpr double _invalid_rate = -1;
-    
-    RationalTime _floor() const {
-        return RationalTime {floor(_value), _rate};
+    friend RationalTime operator * (const RationalTime& lhs, float rhs) {
+        return RationalTime(lhs._value * rhs, lhs._rate);
+    }
+    friend RationalTime operator * (const RationalTime& lhs, double rhs) {
+        return RationalTime(lhs._value * rhs, lhs._rate);
     }
 
+    friend RationalTime operator * (float lhs, const RationalTime& rhs) {
+        return RationalTime(rhs._value * lhs, rhs._rate);
+    }
+    friend RationalTime operator * (double lhs, const RationalTime& rhs) {
+        return RationalTime(rhs._value * lhs, rhs._rate);
+    }
```

Comment:
In the code base I'm working on, there was a rather large amount of code that looked like 

```cpp
t2 = RationalTime(t1.value() * 0.5, t1.rate())
```
to the degree that I felt it was really obscuring the intent of the code, and once again, I encountered some bugs with hardcoded framerates, or reusing framerates from other time objects. I added some local helpers while sorting out the code, and I used them so often, I thought I would promote them to the API as others will probably hit them.

For context, I've been porting the time-related parts of a large QuickTime based application to opentime, and many of the "shenanigans" I've been encountering existed as implementation bugs against the QuickTime SDK. So as I did the port, I was thinking, how would I change the API to avoid these bugs in the future, and that's what led to the introduction of rule-of-5, the maths, and a few additional operators.

Reply:
I only mentioned it because I had some recollection of `RationalTime` scalars being intentionally omit in the past.

For reference, the current API suggestion for the example you gave above is to do this:
```python
half_xform = TimeTransform(scale=0.5)
t2 = half_xform.applied_to(t1)
```
But every time I do that it feels a little clumsy.

I think `TimeTransform` is useful in instances where you're applying both offset and scale or working with `TimeRange`.
So I agree being able to do:
```python
t2 = t1 * 0.5
```
Is a much clearer solution for the 80% use case with `RationalTime`. I just wanted to see if @ssteinbach had a memory of why we omit the scalar operator originally.

Reply:
Marking as resolved, as no further comments have been posted on the reason the scalar operator was omitted.